markdown.go

```diff
@@ -639,9 +639,6 @@ func scanLinkRef(p *parser, data []byte, i int) (linkOffset, linkEnd, titleOffse
 	for i < len(data) && data[i] != ' ' && data[i] != '\t' && data[i] != '\n' && data[i] != '\r' {
 		i++
 	}
-	if i == len(data) {
-		return
-	}
```

Comment:
Just a thought, but do you know of any preconditions that must be met when calling `scanLinkRef`? If so, it would be tremendously helpful to document them. That way, it may become more viable to verify that the code inside does the right thing on all input.

As it stands now, without documentation, this func can accept arbitrary input, and it's not very clear what the exact responsibility/postconditions this func is expected to produce.